README.md

```diff
@@ -24,91 +24,141 @@ The `laundry` HTTP server provides an REST API and online tool to try out the co
 
 Conversions are performed in single-use disposable Docker containers. The containers are secured, and their runtime is [gVisor](https://gvisor.dev/) `runsc`. It provides an additional layer of isolation for the containers.
 
-## Building
+## Installation instructions
 
-To rebuild the uberjar and docker images, run:
+Three kinds of installation methods are presented in the following subsections. 
 
-    ./rebuild.sh
+- We recommend to start by doing a [Temporary installation for demo purposes](#temporary-installation-for-demo-purposes) so that you can integrate the `laundry` to your systems and processes. 
+- The [Production installation with Docker and gVisor runsc](#production-installation-with-docker-and-gvisor-runsc) outlines the procedures for a common installation on a dedicated server or vm.
+- You can optionally continue to harden the setup with instructions given in [Customized production  installations](#customized-production-installations)
 
-To rebuild just the uberjar:
+### Temporary installation for demo purposes
 
-    lein uberjar
+**System requirements:** Linux or Mac & Docker
 
-## Development environment
+This installation method gives you the option to try out the laundry, integrate it into your systems or just to play around with it. This temporary installation method is **not suitable for production use** as it lacks sandboxing.
 
-    java -jar target/uberjar/laundry.jar
+```sh
+git clone https://github.com/solita/laundry.git
+./laundry/docker-demo/build-and-run.sh
+```
 
-Access swagger API docs on http://localhost:9001/api-docs/
+The script builds the necessary docker images including a temporary `laundry-demo`. It starts a docker container for the `laundry` HTTP server. The Docker host socket is exposed to the container, so that the `laundry-demo` can create temporary sibling containers for each conversion. `gVisor runsc` runtime is **not used** in the demo installation.
 
-## Vagrant environment
+Default port is `8080`. The port can be given as parameter to the script
 
-Vagrant uses ansible playbooks under `ansible/` to build (first time) and deploy laundry and the docker images to the vagrant box.
+    ./docker-demo/build-and-run.bash -p 7777
 
-    vagrant up --provision
+See the script output for random api-key and the HTTP API address. Exit the demo with `docker stop laundry-demo`.
 
-By default the host port 8080 (or the first available port after it) is forwarded to the guest, so to access swagger API docs, open <http://localhost:8080/api-docs/>.
+**Note:** Windows Subsystem for Linux users should be able to use the provided scripts. This has been tested on WSL version 2.
 
-Username for the API is `laundry-api`. The password can be read with the following commands:
+**Note:** macOS users might need to edit the script to run the `laundry-demo` container with `--user=root`, because the Docker socket has `root:root` ownership in the container.
```

Comment:
```suggestion
**Note:** macOS users might need to edit the script to run the `laundry-demo` container with `--user=root`, because the Docker socket has `root:root` ownership in the container.

**Note:** The demo configures Docker to expose this port to the internet and may open the host firewall for it.
```